logic/resolvable/Concludable.java

```diff
@@ -84,23 +84,15 @@ private Concludable(Conjunction conjunction) {
     }
 
     public ResourceIterator<Rule> getApplicableRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-        if (applicableRules == null) computeApplicableRules(conceptMgr, logicMgr);
+        synchronized (this) {
+            if (applicableRules == null) applicableRules = applicableRules(conceptMgr, logicMgr);
+        }
```

Comment:
this class is shared across threads so the computation of applicable rules needs to be thread safe

Reply:
Now we can remove the `TODO Implement` in the constructor :)